tests/support/proxy.rs

```diff
@@ -9,8 +9,12 @@ pub fn new() -> Proxy {
 pub struct Proxy {
     controller: Option<controller::Listening>,
     identity: Option<controller::Listening>,
-    inbound: Option<server::Listening>,
-    outbound: Option<server::Listening>,
+
+    inbound: Option<SocketAddr>,
+    outbound: Option<SocketAddr>,
+
+    inbound_server: Option<server::Listening>,
```

Comment:
I think it would be good to get a comment here explaining what the difference between `{inbound, outbound}` and `{inbound, outbound}_server` is.